src/shared/export-json-file.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Provided an object and a filename, allows to generate a downloadable file
+ * containing a JSON-serialized version of that object
+ *
+ * @param _document - Test seam
+ * @return The contents of the downloaded file
+ */
+export function exportJSONFile(
+  exportData: object,
+  filename: string,
+  _document = document
+): string {
+  const link = _document.createElement('a');
+  const fileContent = JSON.stringify(exportData, null, 2);
```

Comment:
At this point I'm not taking into consideration possible non-serializable content on provided data, like functions and such. I don't think we need to worry about that for now.

Reply:
I don't think we need to "deal" with non-serializable values, as such, but do you think perhaps we could handle the fact that this function can throw? It could be the responsibility of the caller to handle exceptions (e.g. maybe we show a toast error if this export-file function encounters an error/throws), but perhaps we could at least document that the function can throw?